test/routes/src/server/routes/t_login.js

```diff
@@ -60,32 +60,115 @@ describe('login routes', () => {
   });
 
   describe('authenticate', () => {
-    it('should redirect from /login/authenticate to SSO', (done) => {
-      supertest(app)
-        .get('/login/authenticate')
-        .expect(res => {
-          const loc = res.header['location'];
-          if (!loc.startsWith(UBUNTU_SSO_URL)) {
-            throw new Error(
-              `Location header ${loc} does not start with ${UBUNTU_SSO_URL}`);
-          }
-        })
-        .expect(302, done);
+    context('with no options', () => {
+      it('should redirect from /login/authenticate to SSO', (done) => {
+        supertest(app)
+          .get('/login/authenticate')
+          .expect(res => {
+            const loc = res.header['location'];
+            if (!loc.startsWith(UBUNTU_SSO_URL)) {
```

Comment:
Instead of "manual" comparisons and throwing errors it's possible to use regular `expect` as in other (unit) tests.

As in: https://github.com/canonical-ols/build.snapcraft.io/blob/master/test/routes/src/server/routes/t_github-auth.js#L111-L115

```js
.expect((res) => {
  expect(res.statusCode).toEqual(302);
  expect(res.headers.location).toEqual('/');
});
```

Not a blocking change, but can make some of the checks more readable.
Also applies to rest of the tests in the file.